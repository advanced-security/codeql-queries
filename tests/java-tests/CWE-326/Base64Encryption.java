import java.io.IOException;
import java.io.PrintWriter;
import java.util.Base64;
import javax.servlet.ServletException;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import javax.servlet.http.HttpSession;

public class Base64Encryption extends HttpServlet {
    @Override
    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
        HttpSession session = request.getSession();
        String requestBody = request.getReader().readLine();
        session.setAttribute("username", requestBody);
        // Do something with the request body here
        String attr = (String)session.getAttribute("username");
        String responseBody = Base64.getEncoder().encodeToString(attr.getBytes());
        // String responseBody = "Encoded username: " + encodedUsername;
        response.setContentType("text/plain");
        response.setCharacterEncoding("UTF-8");
        response.setStatus(HttpServletResponse.SC_OK);
        PrintWriter out = response.getWriter();
        out.print(responseBody);
        out.flush();
    }
}