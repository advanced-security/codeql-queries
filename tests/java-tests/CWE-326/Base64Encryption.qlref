CWE-326/Base64Encryption.ql