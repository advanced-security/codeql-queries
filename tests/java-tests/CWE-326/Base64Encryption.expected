edges
| Base64Encryption.java:17:23:17:62 | (...)... : String | Base64Encryption.java:18:66:18:69 | attr : String |
| Base64Encryption.java:17:31:17:62 | getAttribute(...) : Object | Base64Encryption.java:17:23:17:62 | (...)... : String |
| Base64Encryption.java:18:66:18:69 | attr : String | Base64Encryption.java:18:66:18:80 | getBytes(...) : byte[] |
| Base64Encryption.java:18:66:18:80 | getBytes(...) : byte[] | Base64Encryption.java:18:31:18:81 | encodeToString(...) |
nodes
| Base64Encryption.java:17:23:17:62 | (...)... : String | semmle.label | (...)... : String |
| Base64Encryption.java:17:31:17:62 | getAttribute(...) : Object | semmle.label | getAttribute(...) : Object |
| Base64Encryption.java:18:31:18:81 | encodeToString(...) | semmle.label | encodeToString(...) |
| Base64Encryption.java:18:66:18:69 | attr : String | semmle.label | attr : String |
| Base64Encryption.java:18:66:18:80 | getBytes(...) : byte[] | semmle.label | getBytes(...) : byte[] |
subpaths
#select
| Base64Encryption.java:18:31:18:81 | encodeToString(...) | Base64Encryption.java:17:31:17:62 | getAttribute(...) : Object | Base64Encryption.java:18:31:18:81 | encodeToString(...) | Sensitive data is being 'encrypted' with Base64 Encoding: $@ | Base64Encryption.java:17:31:17:62 | getAttribute(...) | user-provided value |
